### PR TITLE
Angular/Blazor integration of filterable Select

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

- #1273 

## 👩‍💻 Implementation

Typical Angular/Blazor integration implementation.

## 🧪 Testing

Normal Angular/Blazor integration unit testing.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by atmgrifter00** [comment]
@m-akinc you mind buddying this?

File: packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleSelectTests.cs [hunk 1]
```diff
@@ -77,6 +77,23 @@ public void DropdownAppearance_AttributeIsSet(DropdownAppearance value, string e
...
+        Assert.Contains(expectedContents, breadcrumb.Markup);
+    }
+
+    [Fact]
+    public void Select_FilterModeNoneIsSet()
+    {
+        var breadcrumb = RenderWithPropertySet(x => x.FilterMode, FilterMode.None);
+
+        Assert.DoesNotContain("filter-mode", breadcrumb.Markup);
+    }
```
Comment (m-akinc) [inline packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleSelectTests.cs:95]:
```suggestion
    public void Select_FilterModeStandardIsSet()
    {
        var expectedContents = "filter-mode=\"standard\"";
        var select = RenderWithPropertySet(x => x.FilterMode, FilterMode.Standard);

        Assert.Contains(expectedContents, select.Markup);
    }

    [Fact]
    public void Select_FilterModeNoneIsSet()
    {
        var select= RenderWithPropertySet(x => x.FilterMode, FilterMode.None);

        Assert.DoesNotContain("filter-mode", select.Markup);
    }
```

**Review by m-akinc — APPROVED** [review]

File: packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleSelectTests.cs (hunk 1 above)
Comment (atmgrifter00) [reply packages/nimble-blazor/Tests/NimbleBlazor.Tests/Unit/Components/NimbleSelectTests.cs:95]: Of _course_ I left some copy-pasta! GAH! 🤦‍♂️

**Review by rajsite — APPROVED** [review]

**Comment by rajsite** [comment]
FYI @msmithNI your review is required as owner

File: angular-workspace/projects/ni/nimble-angular/src/directives/select/tests/nimble-select.directive.spec.ts [hunk 2]
```diff
@@ -116,6 +122,11 @@ describe('Nimble select', () => {
             expect(nativeElement.appearance).toBe(DropdownAppearance.block);
         });
 
+        it('will use template string values for filterMode', () => {
+            expect(directive.filterMode).toBe(FilterMode.standard);
+            expect(directive.filterMode).toBe(FilterMode.standard);
```
Comment (msmithNI) [inline angular-workspace/projects/ni/nimble-angular/src/directives/select/tests/nimble-select.directive.spec.ts:127]: Looks like this line should be checking `nativeElement` not `directive`

**Review by msmithNI — APPROVED** [review]

**Pull request merged**